common/djangoapps/third_party_auth/views.py

```diff
@@ -61,6 +61,16 @@ def lti_login_and_complete_view(request, backend, *args, **kwargs):
     return complete(request, backend, *args, **kwargs)
 
 
+def login(*args, **kwargs):
+    """
+    Wraps the python social auth login view to return a 404 if third party
+    auth is disabled.
+    """
+    if not SAMLConfiguration.is_enabled():
+        raise Http404
```

Comment:
This view is used by other types of third party auth, not just SAML. So this fix would break things like Google logins (`/auth/login/google-oauth2/`) which don't depend on SAMLConfiguration.